micro_sam/evaluation/inference.py

```diff
@@ -392,118 +389,127 @@ def _save_segmentation(masks, prediction_path):
     shape = masks.shape[-2:]
     masks = [{"segmentation": mask, "area": mask.sum()} for mask in masks]
     segmentation = mask_data_to_segmentation(masks, shape, with_background=True)
-    imageio.imwrite(prediction_path, segmentation)
+    imageio.imwrite(prediction_path, segmentation, compression=5)
+
+
+def extract_instances_from_batched_outputs(batched_outputs):
+    masks = [m["segmentation"][None] for m in batched_outputs]
+    masks = torch.stack(masks)
+    masks = (masks > 0.5).to(torch.float32)
+    return masks
+
+
+def get_channelwise_gt_objects(gt, gt_ids, device=None):
+    device = util._get_device(device)
+    sampled_binary_gt = np.stack([(gt == idx)[None] for idx in gt_ids])
+    sampled_binary_gt = torch.from_numpy(sampled_binary_gt).to(torch.float32)
+    sampled_binary_gt = sampled_binary_gt.to(device)
+    return sampled_binary_gt
 
 
+@torch.no_grad()
 def _run_inference_with_iterative_prompting_for_image(
-    model,
+    predictor,
     image,
     gt,
-    n_iterations,
-    device,
+    use_points,
     use_boxes,
-    prediction_paths,
+    dilation,
     batch_size,
-):
-    assert len(prediction_paths) == n_iterations, f"{len(prediction_paths)}, {n_iterations}"
-    to_sam_inputs = ConvertToSamInputs()
+    embedding_path,
+    n_iterations,
+    prediction_paths
+) -> None:
+    prompt_generator = IterativePromptGenerator()
 
-    image = torch.from_numpy(
-        image[None, None] if image.ndim == 2 else image[None]
+    gt_ids = np.unique(gt)[1:]
+    n_positives = 1 if not use_boxes else 0
```

Comment:
You can simplify this a bit once we have the single boolean `start_with_box_prompt`:
```python
n_negatives = 0
if start_with_box_prompt:
  use_boxes, use_points = True, False
  n_positives = 0
  multimasking = False
else:
  use_boxes, use_points = False, True
  n_positives = 0
  multimasking = True
```

Reply:
See also below for comment w.r.t. multi-masking.